syntax = "proto3";

service Storage {
    rpc StoreContent(Content) returns (SnapshotID){}
    rpc GetContent(SnapshotID) returns (Content){}
}

message Content {
    string url = 1;
    string timestamp = 2;
    bytes data = 3;
}

message SnapshotID {
    string id = 1;
}
